declarations/types.d.ts

```diff
@@ -0,0 +1,132 @@
+    show_on_profile: boolean;
+    user_tags: string[];
+}> {
+}
+export declare type Location = {
+    city: string | null;
+    country: string | null;
+    name: string | null;
+    position: {
+        latitude: number | null;
+        longitude: number | null;
+    };
+};
+export interface BaseVeryBasic extends ForExtendHelper<Entity & {
+    created_at: string;
+    updated_at: string;
+    urls: Urls;
+}> {
+}
+export interface BaseBasic extends ForExtendHelper<BaseVeryBasic & {
+    promoted_at: string | null;
+    links: {
+        html: string;
+        download: string;
+    };
+    width: number;
+    height: number;
+    description: string | null;
+    alt_description: string | null;
+    color: string | null;
+    likes: number;
+    liked_by_user: boolean;
+    categories: {}[];
+}> {
+}
+export interface BaseFull extends ForExtendHelper<{
+    exif: Exif;
+    location: Location;
+    related_collections: {
+        type: RelatedCollectionsType.related | RelatedCollectionsType.collected;
+        total: number;
+    };
+    meta: {
+        index: boolean;
+    };
+}> {
+}
+export interface VeryBasic extends ForExtendHelper<BaseVeryBasic> {
+}
+export interface Basic extends ForExtendHelper<BaseBasic & {
+    user: unknown;
+    current_user_collections?: unknown[];
+    sponsorship: unknown | null;
+}> {
+}
+export interface BasicWithTags extends ForExtendHelper<Basic & {
+    tags: unknown[];
+}> {
+}
+export interface Full extends ForExtendHelper<BaseFull & BasicWithTags & {
+    related_collections: {
+        results: unknown[];
+    };
+}> {
+}
+export declare type Union = InclusiveOr<VeryBasic | BasicWithTags | WithOrWithout<Basic | Full, OwnerOrAdminInfo | OwnerInfo>, Stats>;
+export declare const checkIsVeryBasic: <T extends InclusiveOr<VeryBasic | Basic | BasicWithTags | Full | (Basic & OwnerOrAdminInfo) | (Basic & OwnerInfo) | (Full & OwnerOrAdminInfo) | (Full & OwnerInfo), Stats>>(entity: T) => entity is T & VeryBasic;
+export declare const union: Union;
+export declare const veryBasicOption: Option.Option<VeryBasic | (VeryBasic & Entity & {
+    statistics: EntityStats;
+}) | (Basic & VeryBasic) | (BasicWithTags & VeryBasic) | (Full & VeryBasic) | (Basic & OwnerOrAdminInfo & VeryBasic) | (Basic & OwnerInfo & VeryBasic) | (Full & OwnerOrAdminInfo & VeryBasic) | (Full & OwnerInfo & VeryBasic) | (Entity & {
+    statistics: EntityStats;
+} & VeryBasic) | (Basic & Entity & {
+    statistics: EntityStats;
+} & VeryBasic) | (BasicWithTags & Entity & {
+    statistics: EntityStats;
+} & VeryBasic) | (Full & Entity & {
+    statistics: EntityStats;
+} & VeryBasic) | (Basic & OwnerOrAdminInfo & Entity & {
+    statistics: EntityStats;
+} & VeryBasic) | (Basic & OwnerInfo & Entity & {
+    statistics: EntityStats;
+} & VeryBasic) | (Full & OwnerOrAdminInfo & Entity & {
+    statistics: EntityStats;
+} & VeryBasic) | (Full & OwnerInfo & Entity & {
+    statistics: EntityStats;
+} & VeryBasic)>;
```

Comment:
Any ideas how to fix this one?

Reply:
This line suppse to do so, but theres unclear typescript error
https://github.com/Bnaya/ts-huge-and-slow-declaration-emit/blob/tricky-trick/src/types.ts#L131
Maybe with addtional tweaks it will work

Reply:
"An interface can only extend an object type or intersection of object types with statically known members.(2312)"
typescript does not like WithOrWithout:
```ts
type ForExtendHelper<T> = T;
type WithOrWithout<A, B> = A | (A & B);
interface Foo extends ForExtendHelper<WithOrWithout<{a: number}, { b: string }>> {

}
```
This also not working:
```ts
type A = { a: number } | { b: string };
interface Bar extends A {};
```
https://www.typescriptlang.org/play?ssl=2&ssc=31&pln=2&pc=28#code/C4TwDgpgBAYg9gJwKIA9gQHYBMASEA2kCAPACoB8UAvFKQNwBQokUA6gJbAAWA8gh9zgBXYMQCCAGigAhSjTFQAPlAAUCgGQyAlI3YZ0CAGYBDAMbQ9BgLbss7Y+gBMUCGkxYAzrESp02PIQQJAK8-JxcwqIA3sYAXFAYQlYARkEAvlJRUMnxHsAIegDmUGnklFEMDGlAA